components/ModIcons.tsx

```diff
@@ -0,0 +1,109 @@
+// Bitset information referenced from the osu-web repository
+// https://github.com/ppy/osu-web/blob/master/app/Singletons/Mods.php
+
+import Image from "next/image";
+import { Tooltip } from "@/components/Tooltip";
+
+const LEGACY_BITSET = [
+  [1 << 0, 'NF'],
+  [1 << 1, 'EZ'],
+  [1 << 2, 'TD'],
+  [1 << 3, 'HD'],
+  [1 << 4, 'HR'],
+  [1 << 5, 'SD'],
+  [1 << 6, 'DT'],
+  [1 << 7, 'RX'],
+  [1 << 8, 'HT'],
+  [1 << 9, 'NC'],
+  [1 << 10, 'FL'],
+  [1 << 12, 'SO'],
+  [1 << 13, 'AP'],
+  [1 << 14, 'PF'],
+  [1 << 20, 'FI'],
+  [1 << 21, 'RD'],
+  [1 << 29, 'V2'],
+  [1 << 30, 'MR'],
+  // mania keys (converts)
+  [1 << 15, '4K'],
+  [1 << 16, '5K'],
+  [1 << 17, '6K'],
+  [1 << 18, '7K'],
+  [1 << 19, '8K'],
+  [1 << 24, '9K'],
+];
+
+const modNames: Record<string, string> = {
+  'NF': 'No Fail',
+  'EZ': 'Easy',
+  'TD': 'Touch Device',
+  'HD': 'Hidden',
+  'HR': 'Hard Rock',
+  'SD': 'Sudden Death',
+  'DT': 'Double Time',
+  'RX': 'Relax',
+  'HT': 'Half Time',
+  'NC': 'Nightcore',
+  'FL': 'Flashlight',
+  'SO': 'Spun Out',
+  'AP': 'Autopilot',
+  'PF': 'Perfect',
+  'FI': 'Fade In',
+  'RD': 'Random',
+  'V2': 'Score V2',
+  'MR': 'Mirror',
+  // mania keys
+  '4K': '4 Keys',
+  '5K': '5 Keys',
+  '6K': '6 Keys',
+  '7K': '7 Keys',
+  '8K': '8 Keys',
+  '9K': '9 Keys',
+};
```

Comment:
Clearly an API problem, by the way, since we shouldn't implement this "glue types" by itself.
 
Going to make a PR on backend so we could work with `Mods` object, but before it, I need to fork the osu library we use on backend, since it seems to miss `Mirror` mod (:

For not, please leave it as is.